src/errors.rs

```diff
@@ -231,3 +231,22 @@ impl Termination for BinstallError {
         code
     }
 }
+
+impl From<std::io::Error> for BinstallError {
+    fn from(err: std::io::Error) -> Self {
+        if err.get_ref().is_some() {
+            let kind = err.kind();
+
+            let inner = err
+                .into_inner()
+                .expect("err.get_ref() returns Some, so err.into_inner() should also return Some");
```

Comment:
This `expect` here should be fine, since we enabled LTO, the compiler should be able to inline `std::io::Error::get_ref` and ``std::io::Error::into_inner` and eliminate the `expect` here as dead code.

Reply:
But I haven't verified it though.